---
title: "Metabolomics"
author: "Johannes Rainer<br><strong>Eurac Research</strong>, Bolzano, Italy<br>johannes.rainer@eurac.edu github/twitter: jotsetung"
date: "12 July 2018"
output: 
  ioslides_presentation:
    widescreen: true
    fig_width: 7
    fig_height: 5
    fig_retina: 2
    fig_caption: false
    transition: faster
    css: jostyle.css
---

<style type="text/css">

slides > slide:not(.nobackground):after {
  content: '';
}

slides > slide {
    -webkit-transition:none !important;transition:none !important;
}

.build > * {
  -webkit-transition: opacity 0.1s ease-in-out;
  -webkit-transition-delay: 0.1s;
  -moz-transition: opacity 0.1s ease-in-out 0.1s;
  -o-transition: opacity 0.1s ease-in-out 0.1s;
  transition: opacity 0.1s ease-in-out 0.1s;
}

</style>

## Content

- Introduction to metabolomics
- Preprocessing of LC-MS data in Bioconductor

## Metabolomics? {.build}

<div>
- Large-scale study of small molecules (metabolites) in a system (cell, tissue,
  organism).
- Metabolites: intermediates and products of cellular processes.
</div>
- Metabolome?
  - **Genome**: what can happen.
  - **Transcriptome**: what appears to be happening.
  - **Proteome**: what makes it happen.
  - **Metabolome**: what actually happened.

<div>
- Metabolome influenced by genetic **and** environmental factors.
</div>

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_01.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_02.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_03.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_04.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_06.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_10.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_11.png")
```

## Metabolites? Metabolomics?

- [Glycolysis](https://en.wikipedia.org/wiki/Glycolysis)

```{r out.width = "800px", echo = FALSE}
knitr::include_graphics("images/glycolysis_12.png")
```


## How can we measure metabolites? {.dark}


- Nuclear Magnetic Resonance (NMR) - not covered here.
- Mass spectrometry (MS)-based metabolomics.
- Targeted metabolomics TODO describe
- Untargeted metabolomics TODO describe


## Mass Spectrometry (MS)

<div style="position:absolute; top:28%; left:7%; width:400px">
```{r out.width = "400px", echo = FALSE}
knitr::include_graphics("images/MS_ionization_mod.png")
```
</div>

## Mass Spectrometry (MS)

<div style="position:absolute; top:28%; left:7%; width:400px">
```{r out.width = "400px", echo = FALSE}
knitr::include_graphics("images/MS_ionization_mod.png")
```
</div>
<div style="position:absolute; top:20%; left:53%; width:300px">
```{r out.width = "300px", echo = FALSE}
knitr::include_graphics("images/Spectrum1_mod.png")
```
</div>

<div style="position:absolute; top:65%;left:7%">
> - **Problem:** unable to distinguish between metabolites with the same 
    mass-to-charge ratio (m/z).
> - **Solution:** separate metabolites prior to MS by another property.
</div>
  
<!-- Now comes the incremental LC-MS description -->
## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/MS_Spectrum1.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum1.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum2.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum3.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum4.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum5.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum6.png")
```

## | **Liquid Chromatography Mass Spectrometry (LC-MS)** {.notransition}

```{r out.width = "700px", echo = FALSE}
knitr::include_graphics("images/LCMS_Spectrum6.png")
```

## LC separation

- list some possibilities how metabolites could be separated.
- Example HILIC.

## LC-MS data preprocessing

> - Chromatographic peak detection
> - Alignment
> - Correspondence

## Chromatographic peak detection

- Aim: identify chromatographic peaks in the data.

```{r out.width = "500px", echo = FALSE}
knitr::include_graphics("images/LCMS-data.png")
```

## Chromatographic peak detection

- Aim: identify chromatographic peaks in the data.

```{r out.width = "500px", echo = FALSE}
knitr::include_graphics("images/LCMS-data-peaks.png")
```

## Chromatographic peak detection

- Aim: identify chromatographic peaks in the data.

```{r out.width = "500px", echo = FALSE}
knitr::include_graphics("images/LCMS-data-peaks.png")
```
<div style = "position:absolute; left:520px; top:300px;">
- allow different rt-widths
- allow some scattering on m/z
- identify *correct* boundaries
</div>

## Chromatographic peak detection

- __centWave__ [Tautenhahn et al. *BMC Bioinformatics*, 2008]:
- Step 1: identify *regions of interest*.

```{r out.width = "500px", echo = FALSE}
knitr::include_graphics("images/centWave-ROI.png")
```

## Chromatographic peak detection

- Step 2: peak detection using continuous wavelet transform.
- Allows detection of peaks with different widths.

```{r out.width = "500px", echo = FALSE}
knitr::include_graphics("images/centWave-CWT.png")
```

## Chromatographic peak detection {.build}

- After reading the data with `readMSData` (`MSnbase` package):
- `xcms`: `findChromPeaks` function, passing settings along with an
  algorithm-specific *parameter* object.

```{r echo = FALSE, warning = FALSE, message = FALSE}
library(xcms)
library(doParallel)
registerDoParallel(2)
register(bpstart(DoparParam()), default = TRUE) 
fls <- dir(system.file("sciex", package = "msdata"), full.names = TRUE)

## Define a data.frame with additional information on the files.
pd <- data.frame(file = basename(fls), injection_idx = c(1, 19),
                 sample = c("POOL_1", "POOL_2"), group = "POOL")
## Read the data
data <- readMSData(fls, pdata = new("NAnnotatedDataFrame", pd), 
                   mode = "onDisk")
data <- pickPeaks(smooth(data, method = "SavitzkyGolay", halfWindowSize = 6))
```

```{r message = FALSE}
cwp <- CentWaveParam(peakwidth = c(2, 10), snthresh = 5)
data <- findChromPeaks(data, param = cwp)
head(chromPeaks(data), n = 3)
```



## Alignment {.build}

- Chromatography subject to (random and systematic) noise.
- Same analyte may elute at slightly different time.

```{r echo = FALSE, out.width = "370px"}
knitr::include_graphics("images/alignment.png")
```

- Shifts are LC-setup dependent, seem to be also analyte dependent.


## Alignment

- How strong the shifts are depends on the LC-setup.

- Many algorithms available [Smith et al. *Brief Bioinformatics* 2013]

> - Main assumption: analytes elute in the same order.

> - `xcms`: `adjustRtime` function with `PeakDensityParam` [Smith et al. *Anal. chem.*
    2006] or `ObiwarpParam` [Prince et al. *Anal. chem.* 2006].


## Correspondence {.build}

- Aim: group peaks across samples, assuming they represent the same ion.
- Depends on proper alignment.

```{r echo = FALSE, out.width = "700px"}
knitr::include_graphics("images/correspondence2_01.png")
```

## Correspondence

- Aim: group peaks across samples, assuming they represent the same ion.
- Depends on proper alignment.

```{r echo = FALSE, out.width = "700px"}
knitr::include_graphics("images/correspondence2_02.png")
```

## Correspondence

- Aim: group peaks across samples, assuming they represent the same ion.
- Depends on proper alignment.

```{r echo = FALSE, out.width = "700px"}
knitr::include_graphics("images/correspondence2_03.png")
```


## Correspondence {.build}

- `xcms`: `groupChromPeaks` with `NearestPeaksParam` [Katajamaa et
  al. *Bioinformatics* 2006] and `PeakDensityParam` [Smith et al. *Anal. chem.*
  2006].

<div>
- *peak density* approach:

```{r echo = FALSE, out.width = "700px"}
knitr::include_graphics("images/correspondence2_density.png")
```
</div>

## Normalization

TODO add content:

- what sources of variation?

- Crucial to add QC controls in the experiment.

- available methods (RUV, linear models).

- MS runs not very expensive, running replicates, QC controls etc usually not a
  problem.

## Identification

- Match compounds based on features' m/z.

- Lab-internal databases with approximate retention times for specific
  compounds.

TODO add content:

- What databases available

- up and coming: compound db, similar to `ensembldb` and alike. Problem: unclear
  license situation.

## Afternoon lab

- LC-MS data handling (`MSnbase`).
- LC-MS data preprocessing using `xcms`.
